app/controllers/projects_controller.rb

```diff
@@ -1,9 +1,51 @@
 class ProjectsController < ApplicationController
   def new
-
+    @project = Project.new
   end
 
   def create
+    @project = Project.new(project_params)
+    @project.user = current_user
+    if @project.save
+      redirect_to project_path(current_user)
+    else
+      render :new
+    end
+  end
+
+  def show
+    @project = Project.find(params[:id])
+  end
+
+  def edit
+    @project = Project.find(params[:id])
+  end
+
+  def update
+    @project = Project.find(params[:id])
+    @project.update(project_params)
+
+    if @project.save
+      redirect_to project_path(current_user)
+    else
+      render :edit
+    end
+  end
+
+  def add
+    @project = Project.find(params[:id])
+    @project.project_users.build(user: current_user, project: @project)
+
+    if @project.save
+      redirect_to @project
+    else
+      render 'show'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.